build fails with GCC 15

### Problem

Neovim fails to build with GCC 15.
I've tested with master (b51110f4a11af114401e626cd4c1f1aec23e81c5), v0.10.3 and v0.10.2.

The following is a snippet from the build error message.
```
FAILED: src/nvim/CMakeFiles/nvim_bin.dir/api/window.c.o
/usr/bin/cc -DHAVE_UNIBILIUM -DINCLUDE_GENERATED_DECLARATIONS -DUNIT_TESTING -D_GNU_SOURCE -Dnvim_bin_EXPORTS -I/home/bdas/neovim/build/src/nvim/auto -I/home/bdas/neovim/bui
ld/include -I/home/bdas/neovim/build/cmake.config -I/home/bdas/neovim/src -isystem /home/bdas/neovim/.deps/usr/include/luajit-2.1 -isystem /home/bdas/neovim/.deps/usr/includ
e -O2 -g -Og -g -flto=auto -fno-fat-lto-objects -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wvla -Wdouble-promotion -
Wmissing-noreturn -Wmissing-format-attribute -Wmissing-prototypes -fsigned-char -fstack-protector-strong -Wno-conversion -fno-common -Wno-unused-result -Wimplicit-fallthroug
h -fdiagnostics-color=always -MD -MT src/nvim/CMakeFiles/nvim_bin.dir/api/window.c.o -MF src/nvim/CMakeFiles/nvim_bin.dir/api/window.c.o.d -o src/nvim/CMakeFiles/nvim_bin.di
r/api/window.c.o -c /home/bdas/neovim/src/nvim/api/window.c
/home/bdas/neovim/src/nvim/api/window.c: In function ‘nvim_win_text_height’:
/home/bdas/neovim/src/nvim/api/window.c:531:73: error: passing argument 4 of ‘normalize_index’ from incompatible pointer type [-Wincompatible-pointer-types]
  531 |     start_lnum = (linenr_T)normalize_index(buf, opts->start_row, false, &oob);
      |                                                                         ^~~~
      |                                                                         |
      |                                                                         _Bool *
In file included from /home/bdas/neovim/src/nvim/api/private/helpers.h:186,
                 from /home/bdas/neovim/src/nvim/api/window.c:9:
/home/bdas/neovim/build/include/api/private/helpers.h.generated.h:31:96: note: expected ‘unsigned int *’ but argument is of type ‘_Bool *’
   31 | DLLEXPORT int64_t normalize_index(buf_T *buf, int64_t index, unsigned end_exclusive, unsigned *oob);
      |                                                                                      ~~~~~~~~~~^~~
/home/bdas/neovim/src/nvim/api/window.c:535:69: error: passing argument 4 of ‘normalize_index’ from incompatible pointer type [-Wincompatible-pointer-types]
  535 |     end_lnum = (linenr_T)normalize_index(buf, opts->end_row, false, &oob);
      |                                                                     ^~~~
      |                                                                     |
      |                                                                     _Bool *
/home/bdas/neovim/build/include/api/private/helpers.h.generated.h:31:96: note: expected ‘unsigned int *’ but argument is of type ‘_Bool *’
   31 | DLLEXPORT int64_t normalize_index(buf_T *buf, int64_t index, unsigned end_exclusive, unsigned *oob);
      |                                                                                      ~~~~~~~~~~^~~
[199/416] Building C object src/nvim/CMakeFiles/nvim_bin.dir/api/vim.c.o
```

Complete build log: https://gist.github.com/listout/03224ff371b149e3b8cdf0bde88564a7

### Steps to reproduce

Use a gcc 15 environment
git clone https://github.com/neovim/neovim.git
cd neovim
make CMAKE_BUILD_TYPE=RelWithDebInfo CMAKE_INSTALL_PREFIX=$HOME/.local/nvim

### Expected behavior

Builds successfully with GCC 15

### Nvim version (nvim -v)

v0.11.0-dev-1414+gb51110f4a1

### Vim (not Nvim) behaves the same?

no sure

### Operating system/version

Gentoo with GCC 15

### Terminal name/version

alacritty

### $TERM environment variable

xterm-256color

### Installation

Git source

--- a/src/mpack/conv.h
+++ b/src/mpack/conv.h
@@ -14,10 +14,6 @@ typedef mpack_sint32_t mpack_sintmax_t;
 typedef mpack_uint32_t mpack_uintmax_t;
 #endif
 
-#ifndef bool
-# define bool unsigned
-#endif
-
 MPACK_API mpack_token_t mpack_pack_nil(void) FUNUSED FPURE;
 MPACK_API mpack_token_t mpack_pack_boolean(unsigned v) FUNUSED FPURE;
 MPACK_API mpack_token_t mpack_pack_uint(mpack_uintmax_t v) FUNUSED FPURE;

